File: TypeSharpSample/ExampleProject/Model/CompositeTypeTest.cs
using ExampleProject.Model.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using TypeSharp;

namespace ExampleProject.Model
{
    [TypeSharpClass]
    public class CompositeTypeTest
    {
        public IUser User { get; set; }
        public DataRepository Repository { get; set; }

        public IDataItem ComplexMethod(string item)
        {
            return null;
        }
    }
}

File: TypeSharpSample/TypeSharpManager/ExampleProject/Model/CompositeTypeTest.ts
/// <reference path="ExampleProject.Model.d.ts" />

module ExampleProject.Model {

	export class CompositeTypeTest {
	
		User: IUser;
	
		Repository: ExampleProject2.Repository.DataRepository;
	
		ComplexMethod(item: string): ExampleProject.Model.Data.IDataItem {
			return null;
		}
	
	}
	
}